### PR TITLE
Feature/improves stop chat

### PR DESCRIPTION
**Pull request by lopezi**

### CODE REVIEW

File: src/DeVinci_frontend/components/ChatBox.svelte [hunk 1]
```diff
@@ -197,17 +220,6 @@
     showToast = false;
   }
 
-
-  function interruptMessageGeneration(): void {
```
Comment (patnorris) [inline src/DeVinci_frontend/components/ChatBox.svelte:201]: I think this function title well describes the intention of the stop chat button functionality

File: src/DeVinci_frontend/components/ChatBox.svelte [hunk 2]
```diff
@@ -41,6 +43,27 @@
 		return { update: scroll }
 	};
 
+  async function showNewChat() {
+    if ($activeChatGlobal !== null) {
```
Comment (patnorris) [inline src/DeVinci_frontend/components/ChatBox.svelte:47]: this shouldn't be included in the function as it starts a completely new chat

File: src/DeVinci_frontend/components/ChatBox.svelte [hunk 3]
```diff
@@ -41,6 +43,27 @@
...
+      try {
+        await $chatModelGlobal.interruptGenerate();
+      } catch (error) {
+        console.error("Error stopping the answer generation on starting new chat ", error);
+      }
+    }
+    messages = [];
+    newMessageText = '';
+    messageGenerationInProgress = false;
+    if (location.hash !== "#/") {
```
Comment (patnorris) [inline src/DeVinci_frontend/components/ChatBox.svelte:62]: this shouldn't be included in the function as we're already in this location and shouldn't change routes

Comment (lopezi) [reply src/DeVinci_frontend/components/ChatBox.svelte]: got it. simplified version in the commit below.

Comment (patnorris) [reply src/DeVinci_frontend/components/ChatBox.svelte]: looks good, thanks

**Review by patnorris — APPROVED** [review]

**Pull request merged**